lib/snapshot/BUILD.gn

```diff
@@ -5,7 +5,7 @@
 import("//flutter/lib/snapshot/toolchain.gni")
 import("//flutter/lib/ui/dart_ui.gni")
 
-target_is_fuchsia = target_os == "fuchsia"
+target_is_fuchsia = target_os == "fuchsia" && current_os == target_os
```

Comment:
Yes.  If we were building on Fuchsia and targeting, say, Windows, then we'd want to use the Fuchsia stuff for the host OS.